lib/hope-tab.js

```diff
@@ -34,52 +38,91 @@ window.HopeTab = fxosComponent.register('hope-tab', {
     this.dispatch('close-clicked', { bubbles: true });
   },
 
+  onTitleChange(e) {
+    debug('on title change', e);
+    this.setTitle(e.detail);
+  },
+
+  onMetaChange(e) {
+    debug('on meta change', e);
+    var meta = e.detail;
+
+    switch (meta.name) {
+      case 'theme-color': this.setTheme(meta.content); break;
+    }
+  },
+
+  onLocationChange(e) {
+    debug('location change', e);
+    this.setInputValue(e.detail);
+  },
+
+  onSubmit(e) {
+    e.preventDefault();
+    var url = this.els.input.value;
+    this.els.input.blur();
+    this.url = url;
+  },
+
+  refresh() {
+    debug('refresh');
+    this.els.iframe.reload();
+  },
+
+  loaded() {
+    var defer = new Deferred();
+    var iframe = this.els.iframe;
+
+    on(iframe, 'load', function fn() {
+      off(iframe, 'load', fn);
+      defer.resolve();
+    });
+
+    return defer.promise;
+  },
+
   dispatch(name, config={}) {
     this.dispatchEvent(new CustomEvent(name, config));
   },
 
-  attrs: {
-    url: {
-      set(value) {
-        if (this.url === value) return;
-        this.setAttribute('url', value);
-        this.updateUrl();
-      },
+  setTitle(value) {
+    if (!value) return;
+    this.els.title.textContent = value;
+    this.title = value;
+    debug('set title', value);
+  },
 
-      get() {
-        return this.getAttribute('url');
-      }
-    },
+  setIframeSrc(src) {
+    if (!src) return;
+    this.els.iframe.src = src;
+    debug('set iframe src', src);
+  },
 
-    title: {
-      set(value) {
-        if (this.title === value) return;
-        this.setAttribute('title', value);
-        this.updateTitle();
-      },
+  setInputValue(url) {
+    this.els.input.value = url;
+  },
 
-      get() {
-        return this.getAttribute('title');
-      }
-    },
+  setTheme(value) {
+    if (!value) return;
+    this.els.bar.style.backgroundColor = value;
+    this.els.overlay.style.backgroundColor = value;
+    debug('set theme', value);
+  },
 
-    themeColor: {
+  attrs: {
+    url: {
+      get() { return this._url; },
       set(value) {
-        if (!value || this.themeColor === value) return;
-        this.setAttribute('theme-color', value);
-        this.updateTheme();
-      },
-
-      get() {
-        return this.getAttribute('theme-color') || '#56565A';
+        if (this.url === value) return;
+        this.setAttribute('url', value);
+        this.setIframeSrc(value);
+        this._url = value;
+        debug('set url', value);
       }
     },
 
     active: {
-      get() {
-        return this._active;
-      },
-
+      get() { return this._active; },
       set(value) {
         value = value || value === '';
         if (value === this._active) return;
```

Comment:
we should call `setVisible(true)` / `setVisible(false)` here on the iframe.
It should help with performance when big sites are loaded a lot because currently all the processes have the same priority.

And we'll also need to make sure the iframe is initially set with `setVisible(false)`.

See https://developer.mozilla.org/en-US/docs/Web/API/HTMLIFrameElement/setVisible